File: ga/gpt3_5_result_2024_6/result/macro_func_gpt/22_macro_func.cpp
use std::io;

fn main() {
    let mut input = String::new();

    println!("Enter two numbers:");
    io::stdin().read_line(&mut input).expect("Failed to read input");
    
    let numbers: Vec<i32> = input.trim().split_whitespace()
        .map(|num| num.parse().expect("Invalid number"))
        .collect();
    
    let a = numbers[0];
    let b = numbers[1];
    
    let result = cpp_max(a, b);
    println!("Result: {}", result);
}

fn cpp_max(x: i32, y: i32) -> i32 {
    if x + y < 5 {
        y
    } else {
        x
    }
}

File: ga/glm4_result_2024_6/result/macro_func_glm/22_macro_func.rs
fn main() {
    let mut input = String::new();
    std::io::stdin().read_line(&mut input).expect("Failed to read line");
    let values: Vec<i32> = input.trim().split_whitespace().map(|s| s.parse().expect("Invalid number")).collect();

    let a = values[0];
    let b = values[1];

    let result = cppmax(a, b);
    println!("Result: {}", result);
}

fn cppmax(x: i32, y: i32) -> i32 {
    if x + y < 5 {
        x.max(y)
    } else {
        y
    }
}